First version of Tagging Service Implementation

schandrika opened the pull request:
First version of Tagging Service Implementation. task #288

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/volttron/volttron/1438)
<!-- Reviewable:end -->

craig8 commented:
During build I get the following warnings about highlighting

````
WARNING:sphinx.highlighting:Could not lex literal_block as "json". Highlighting skipped.
/home/vdev/vchecks/docs/source/specifications/tagging_service.rst:339: WARNING: Could not lex literal_block as "json". Highlighting skipped.
WARNING:sphinx.highlighting:Could not lex literal_block as "json". Highlighting skipped.
/home/vdev/vchecks/docs/source/specifications/tagging_service.rst:352: WARNING: Could not lex literal_block as "json". Highlighting skipped.
````

craig8 commented:
Unless you have an exposed rpc function all internal class functions should begin with an _ as those should not show up in the interface of the Tagging Agent @schandrika

craig8 commented on services/core/MongodbTaggingService/mongodb/resources/categories.csv:
```diff
@@ -0,0 +1,27 @@
+name,description
```
Would this and the other "resources" be better placed at a root level of the repository rather than packaged with the agents?

schandrika replied:
I put it in two location since in theory they could be customized - adding new tags, categories etc. Also so that I could use resource_string method from package_resources. what do you think?

craig8 replied:
I think that if the user wants to customize the csv then they should have a place to pass in the categories during runtime...i.e I don't however think that this is very likely to happen.  As with the resource_string option I think the volttron_root path would be more maintainable.

craig8 commented on volttron/platform/agent/base_tagging.py:
```diff
@@ -0,0 +1,1044 @@
+        Called right after setup to load a dictionary of valid tags. It
+        should load self.valid_tags with tag and type information
+
+        """
```
add pass here

schandrika replied:
true. should i just pass that as args or expect the function to return the valid tags

craig8 replied:
I don't understand your response....

craig8 commented on volttron/platform/agent/base_tagging.py:
```diff
@@ -0,0 +1,1044 @@
+        its corresponding parent tag. Implementing methods should load
+        self.tag_refs with tag and parent tag information
+        """
+
```
add pass

craig8 commented on volttron/platform/agent/base_tagging.py:
```diff
@@ -0,0 +1,1044 @@
+
+
+
+'''
```
What is the following code for please add documentation around what it actually is doing?

craig8 commented on volttrontesting/services/tagging/test_tagging.py:
```diff
@@ -0,0 +1,1327 @@
+tagging_service_id = None
+sqlite_config = {"connection": {"type": "sqlite",
+                 "params": {"database":
+                                "~/.volttron/data/volttron.tags.sqlite"}},
```
This should not be hardcoded to be using the user home dir it should be based upon the instances volttron_home during the test.

schandrika commented on services/core/SQLiteTaggingService/sqlite/resources/categories.csv:
```diff
@@ -0,0 +1,27 @@
+name,description
```
I put it in two location since in theory they could be customized - adding new tags, categories etc. Also so that I could use resource_string method from package_resources

craig8 commented on volttron/platform/agent/base_tagging.py:
```diff
@@ -0,0 +1,1044 @@
+        return ujson.loads(data_string, precise_float=True)
+except ImportError:
+    from zmq.utils.jsonapi import dumps, loads
+
```
Note there are now utility functions for serializing using volttron.platform.agent.json see https://github.com/VOLTTRON/volttron/blob/develop/volttron/platform/agent/json.py

schandrika commented:
Review status: 0 of 36 files reviewed at latest revision, 16 unresolved discussions.

---

*[services/core/SQLiteTaggingService/sqlite/tagging.py, line 135 at r1](https://reviewable.io:443/reviews/volttron/volttron/1438#-Kr4tzXEbKQDh1o0Ghxu-r1-135:-KsgUFOy40muD9BHlNEc:b-vwwgza) ([raw file](https://github.com/volttron/volttron/blob/a3a6903e590807b0d45649333cd43bef5b26f7ce/services/core/SQLiteTaggingService/sqlite/tagging.py#L135)):*
<details><summary><i>Previously, kmonson wrote…</i></summary><blockquote>

Is there a reason this is done last? If so it needs a comment explaining why.
</blockquote></details>

will move it up

---

*[volttron/platform/agent/base_tagging.py, line 95 at r1](https://reviewable.io:443/reviews/volttron/volttron/1438#-Kr4tzXEbKQDh1o0Ghxv-r1-95:-KsgWbvdCrotmKyOhxVt:bvnrpcm) ([raw file](https://github.com/volttron/volttron/blob/a3a6903e590807b0d45649333cd43bef5b26f7ce/volttron/platform/agent/base_tagging.py#L95)):*
<details><summary><i>Previously, kmonson wrote…</i></summary><blockquote>

Don't do the * import. Pick out what you need.
</blockquote></details>

updated

---

*[volttron/platform/agent/base_tagging.py, line 106 at r1](https://reviewable.io:443/reviews/volttron/volttron/1438#-Kr4tzXEbKQDh1o0Ghxv-r1-106:-KsgWtZ4itzCIgk1rGeI:b9zrb0f) ([raw file](https://github.com/volttron/volttron/blob/a3a6903e590807b0d45649333cd43bef5b26f7ce/volttron/platform/agent/base_tagging.py#L106)):*
<details><summary><i>Previously, craig8 (Craig) wrote…</i></summary><blockquote>

Note there are now utility functions for serializing using volttron.platform.agent.json see https://github.com/VOLTTRON/volttron/blob/develop/volttron/platform/agent/json.py
</blockquote></details>

Removed the unused imports and defs

---

*[volttron/platform/agent/base_tagging.py, line 110 at r1](https://reviewable.io:443/reviews/volttron/volttron/1438#-Kr4tzXEbKQDh1o0Ghxv-r1-110:-KsgX12S9tqVfXL1EMxy:bt2vr1b) ([raw file](https://github.com/volttron/volttron/blob/a3a6903e590807b0d45649333cd43bef5b26f7ce/volttron/platform/agent/base_tagging.py#L110)):*
<details><summary><i>Previously, kmonson wrote…</i></summary><blockquote>

We should do this in the SQLite tagging service agent as that is only where it applies.
</blockquote></details>

Fixed.

---

*[volttron/platform/agent/base_tagging.py, line 126 at r1](https://reviewable.io:443/reviews/volttron/volttron/1438#-Kr4tzXEbKQDh1o0Ghxv-r1-126:-KsgX3fEYao5m_CzKTjM:b4z38q8) ([raw file](https://github.com/volttron/volttron/blob/a3a6903e590807b0d45649333cd43bef5b26f7ce/volttron/platform/agent/base_tagging.py#L126)):*
<details><summary><i>Previously, kmonson wrote…</i></summary><blockquote>

Is there a specific reason that you call this after you setup your instance variables? Unless these is a good reason you usually do this first.
</blockquote></details>

Moved it up. 

---

*[volttron/platform/agent/base_tagging.py, line 163 at r1](https://reviewable.io:443/reviews/volttron/volttron/1438#-Kr4tzXEbKQDh1o0Ghxv-r1-163:-KsgVg9Am8AT-M7mGlsw:b-7ql1td) ([raw file](https://github.com/volttron/volttron/blob/a3a6903e590807b0d45649333cd43bef5b26f7ce/volttron/platform/agent/base_tagging.py#L163)):*
<details><summary><i>Previously, craig8 (Craig) wrote…</i></summary><blockquote>

add pass
</blockquote></details>

fixed

---

*[volttron/platform/agent/base_tagging.py, line 687 at r1](https://reviewable.io:443/reviews/volttron/volttron/1438#-Kr4tzXEbKQDh1o0Ghxv-r1-687:-KsgWCuYxBUZN5PF0GUG:b-w7lq63) ([raw file](https://github.com/volttron/volttron/blob/a3a6903e590807b0d45649333cd43bef5b26f7ce/volttron/platform/agent/base_tagging.py#L687)):*
<details><summary><i>Previously, kmonson wrote…</i></summary><blockquote>

You probably just need to remove the target cases doc string and make a note that it's ply related stuff.
</blockquote></details>

It is a query api helper method. Users could send query string or a dictionary of key:value and / or condition. If the parameter sent is dict, this method helps to create the user query string, so that rest of the query parsing logic would be the same.  It already has doc string. Does the doc string need more explanation?

---

*[volttron/platform/agent/base_tagging.py, line 832 at r1](https://reviewable.io:443/reviews/volttron/volttron/1438#-Kr4tzXEbKQDh1o0Ghxv-r1-832:-KsgXGTw05ekA5ofmLMa:b-pj7jx1) ([raw file](https://github.com/volttron/volttron/blob/a3a6903e590807b0d45649333cd43bef5b26f7ce/volttron/platform/agent/base_tagging.py#L832)):*
<details><summary><i>Previously, kmonson wrote…</i></summary><blockquote>

Change this to a comment or remove it.
</blockquote></details>

changed to comment

---

*[volttron/platform/agent/base_tagging.py, line 1027 at r1](https://reviewable.io:443/reviews/volttron/volttron/1438#-Kr4tzXEbKQDh1o0Ghxv-r1-1027:-KsgXJyfhKw1fE093Asv:b285qk6) ([raw file](https://github.com/volttron/volttron/blob/a3a6903e590807b0d45649333cd43bef5b26f7ce/volttron/platform/agent/base_tagging.py#L1027)):*
<details><summary><i>Previously, kmonson wrote…</i></summary><blockquote>

Should this go in some tests?
</blockquote></details>

  I added it here initially tojust  visually inspect the query string.  All these cases are currently tested by test_tagging.py the only different between testing it is here vs testing it through the implementing class is that the query is actually executed and hence easier to compare expected vs actual in pytest.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/volttron/volttron/1438)*
<!-- Sent from Reviewable.io -->

schandrika commented:
Review status: 0 of 36 files reviewed at latest revision, 16 unresolved discussions.

---

*[volttrontesting/services/tagging/test_tagging.py, line 86 at r1](https://reviewable.io:443/reviews/volttron/volttron/1438#-Kr4tzXEbKQDh1o0Ghxw-r1-86:-KsgYevfgaD2hwpt5WFt:btedssy) ([raw file](https://github.com/volttron/volttron/blob/a3a6903e590807b0d45649333cd43bef5b26f7ce/volttrontesting/services/tagging/test_tagging.py#L86)):*
<details><summary><i>Previously, craig8 (Craig) wrote…</i></summary><blockquote>

This should not be hardcoded to be using the user home dir it should be based upon the instances volttron_home during the test.
</blockquote></details>

It is getting set dynamically to volttron instance home in tagging_service fixture. Will remove the hardcoded path (that is not used ) from the declaration. 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/volttron/volttron/1438)*
<!-- Sent from Reviewable.io -->

schandrika commented:
Review status: 0 of 36 files reviewed at latest revision, 16 unresolved discussions.

---

*[services/core/MongodbTaggingService/mongodb/tagging.py, line 102 at r1](https://reviewable.io:443/reviews/volttron/volttron/1438#-Kr4tzXEbKQDh1o0Ghxi-r1-102:-KsgZ8PHlAymce-jJnlh:b-hfpv0l) ([raw file](https://github.com/volttron/volttron/blob/a3a6903e590807b0d45649333cd43bef5b26f7ce/services/core/MongodbTaggingService/mongodb/tagging.py#L102)):*
<details><summary><i>Previously, kmonson wrote…</i></summary><blockquote>

I don't think this is necessary any more as the class is declared with it's actual name instead of "Agent".
</blockquote></details>

removed

---

*[services/core/MongodbTaggingService/mongodb/tagging.py, line 138 at r1](https://reviewable.io:443/reviews/volttron/volttron/1438#-Kr4tzXEbKQDh1o0Ghxi-r1-138:-KsgZEefZdo2jo-Myily:b-7mmp82) ([raw file](https://github.com/volttron/volttron/blob/a3a6903e590807b0d45649333cd43bef5b26f7ce/services/core/MongodbTaggingService/mongodb/tagging.py#L138)):*
<details><summary><i>Previously, kmonson wrote…</i></summary><blockquote>

Is there a reason we do this last?
</blockquote></details>

moved

---

*[services/core/MongodbTaggingService/mongodb/resources/categories.csv, line 1 at r1](https://reviewable.io:443/reviews/volttron/volttron/1438#-Kr4tzXEbKQDh1o0Ghxd-r1-1:-KsgZMbD21Ve1FsCidSB:bnmrp19) ([raw file](https://github.com/volttron/volttron/blob/a3a6903e590807b0d45649333cd43bef5b26f7ce/services/core/MongodbTaggingService/mongodb/resources/categories.csv#L1)):*
<details><summary><i>Previously, craig8 (Craig) wrote…</i></summary><blockquote>

I think that if the user wants to customize the csv then they should have a place to pass in the categories during runtime...i.e I don't however think that this is very likely to happen.  As with the resource_string option I think the volttron_root path would be more maintainable.
</blockquote></details>

I will move it to common location

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/volttron/volttron/1438)*
<!-- Sent from Reviewable.io -->

schandrika commented:
Review status: 0 of 36 files reviewed at latest revision, 16 unresolved discussions.

---

*[volttron/platform/agent/base_tagging.py, line 154 at r1](https://reviewable.io:443/reviews/volttron/volttron/1438#-Kr4tzXEbKQDh1o0Ghxv-r1-154:-KsgZYUz6GYNfXv_BSYD:b-tisdqm) ([raw file](https://github.com/volttron/volttron/blob/a3a6903e590807b0d45649333cd43bef5b26f7ce/volttron/platform/agent/base_tagging.py#L154)):*
<details><summary><i>Previously, craig8 (Craig) wrote…</i></summary><blockquote>

I don't understand your response....
</blockquote></details>

Kyle asked me to ignore this comment

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/volttron/volttron/1438)*
<!-- Sent from Reviewable.io -->

schandrika commented:
Review status: 0 of 31 files reviewed at latest revision, 16 unresolved discussions.

---

*[services/core/MongodbTaggingService/mongodb/tagging.py, line 104 at r1](https://reviewable.io:443/reviews/volttron/volttron/1438#-Kr4tzXEbKQDh1o0Ghxi-r1-104:-KskOz5Ki2DEkTHbRmHi:bv8pk7f) ([raw file](https://github.com/volttron/volttron/blob/a3a6903e590807b0d45649333cd43bef5b26f7ce/services/core/MongodbTaggingService/mongodb/tagging.py#L104)):*
<details><summary><i>Previously, kmonson wrote…</i></summary><blockquote>

There is no positional argument and your constructor requires one.
</blockquote></details>

As discussed, kwargs maps based on argument name even if it is a positional argument. Added check for connection details in config and raised ValueError if not present 

---

*[services/core/MongodbTaggingService/mongodb/tagging.py, line 138 at r1](https://reviewable.io:443/reviews/volttron/volttron/1438#-Kr4tzXEbKQDh1o0Ghxi-r1-138:-KskPE8egEKWmMLvm8gP:b-896fix) ([raw file](https://github.com/volttron/volttron/blob/a3a6903e590807b0d45649333cd43bef5b26f7ce/services/core/MongodbTaggingService/mongodb/tagging.py#L138)):*
<details><summary><i>Previously, schandrika wrote…</i></summary><blockquote>

moved
</blockquote></details>

Done.

---

*[services/core/SQLiteTaggingService/sqlite/tagging.py, line 135 at r1](https://reviewable.io:443/reviews/volttron/volttron/1438#-Kr4tzXEbKQDh1o0Ghxu-r1-135:-KskPKWjf2RLft5P8LHL:b-896fix) ([raw file](https://github.com/volttron/volttron/blob/a3a6903e590807b0d45649333cd43bef5b26f7ce/services/core/SQLiteTaggingService/sqlite/tagging.py#L135)):*
<details><summary><i>Previously, schandrika wrote…</i></summary><blockquote>

will move it up
</blockquote></details>

Done.

---

*[volttron/platform/agent/base_tagging.py, line 95 at r1](https://reviewable.io:443/reviews/volttron/volttron/1438#-Kr4tzXEbKQDh1o0Ghxv-r1-95:-KskPMJUl5JObLKZeZIB:b-896fix) ([raw file](https://github.com/volttron/volttron/blob/a3a6903e590807b0d45649333cd43bef5b26f7ce/volttron/platform/agent/base_tagging.py#L95)):*
<details><summary><i>Previously, schandrika wrote…</i></summary><blockquote>

updated
</blockquote></details>

Done.

---

*[volttron/platform/agent/base_tagging.py, line 106 at r1](https://reviewable.io:443/reviews/volttron/volttron/1438#-Kr4tzXEbKQDh1o0Ghxv-r1-106:-KskPOI2wC4qphaBGpWA:b-896fix) ([raw file](https://github.com/volttron/volttron/blob/a3a6903e590807b0d45649333cd43bef5b26f7ce/volttron/platform/agent/base_tagging.py#L106)):*
<details><summary><i>Previously, schandrika wrote…</i></summary><blockquote>

Removed the unused imports and defs
</blockquote></details>

Done.

---

*[volttron/platform/agent/base_tagging.py, line 110 at r1](https://reviewable.io:443/reviews/volttron/volttron/1438#-Kr4tzXEbKQDh1o0Ghxv-r1-110:-KskPNBIwPxkgprk0R5O:b-896fix) ([raw file](https://github.com/volttron/volttron/blob/a3a6903e590807b0d45649333cd43bef5b26f7ce/volttron/platform/agent/base_tagging.py#L110)):*
<details><summary><i>Previously, schandrika wrote…</i></summary><blockquote>

Fixed.
</blockquote></details>

Done.

---

*[volttron/platform/agent/base_tagging.py, line 126 at r1](https://reviewable.io:443/reviews/volttron/volttron/1438#-Kr4tzXEbKQDh1o0Ghxv-r1-126:-KskPQu-Af56q9yLJspS:b-896fix) ([raw file](https://github.com/volttron/volttron/blob/a3a6903e590807b0d45649333cd43bef5b26f7ce/volttron/platform/agent/base_tagging.py#L126)):*
<details><summary><i>Previously, schandrika wrote…</i></summary><blockquote>

Moved it up. 
</blockquote></details>

Done.

---

*[volttron/platform/agent/base_tagging.py, line 154 at r1](https://reviewable.io:443/reviews/volttron/volttron/1438#-Kr4tzXEbKQDh1o0Ghxv-r1-154:-KskPSG3GNt2l8BxrLCi:b-896fix) ([raw file](https://github.com/volttron/volttron/blob/a3a6903e590807b0d45649333cd43bef5b26f7ce/volttron/platform/agent/base_tagging.py#L154)):*
<details><summary><i>Previously, schandrika wrote…</i></summary><blockquote>

Kyle asked me to ignore this comment
</blockquote></details>

Done.

---

*[volttron/platform/agent/base_tagging.py, line 163 at r1](https://reviewable.io:443/reviews/volttron/volttron/1438#-Kr4tzXEbKQDh1o0Ghxv-r1-163:-KskPSrHZrQ8mAm03Arf:b-896fix) ([raw file](https://github.com/volttron/volttron/blob/a3a6903e590807b0d45649333cd43bef5b26f7ce/volttron/platform/agent/base_tagging.py#L163)):*
<details><summary><i>Previously, schandrika wrote…</i></summary><blockquote>

fixed
</blockquote></details>

Done.

---

*[volttron/platform/agent/base_tagging.py, line 832 at r1](https://reviewable.io:443/reviews/volttron/volttron/1438#-Kr4tzXEbKQDh1o0Ghxv-r1-832:-KskPVvmTSCgNc-TOR6F:b-896fix) ([raw file](https://github.com/volttron/volttron/blob/a3a6903e590807b0d45649333cd43bef5b26f7ce/volttron/platform/agent/base_tagging.py#L832)):*
<details><summary><i>Previously, schandrika wrote…</i></summary><blockquote>

changed to comment
</blockquote></details>

Done.

---

*[volttron/platform/agent/base_tagging.py, line 1027 at r1](https://reviewable.io:443/reviews/volttron/volttron/1438#-Kr4tzXEbKQDh1o0Ghxv-r1-1027:-KskPZRgX4xQRw2nztUm:b-896fix) ([raw file](https://github.com/volttron/volttron/blob/a3a6903e590807b0d45649333cd43bef5b26f7ce/volttron/platform/agent/base_tagging.py#L1027)):*
<details><summary><i>Previously, schandrika wrote…</i></summary><blockquote>

  I added it here initially tojust  visually inspect the query string.  All these cases are currently tested by test_tagging.py the only different between testing it is here vs testing it through the implementing class is that the query is actually executed and hence easier to compare expected vs actual in pytest.
</blockquote></details>

Done.

---

*[volttron_data/tagging_resources/categories.csv, line 1 at r1](https://reviewable.io:443/reviews/volttron/volttron/1438#-Kr4tzXEbKQDh1o0Ghxd-r1-1:-KskPalOw2Qef-nbPi_T:b-896fix) ([raw file](https://github.com/volttron/volttron/blob/a3a6903e590807b0d45649333cd43bef5b26f7ce/volttron_data/tagging_resources/categories.csv#L1)):*
<details><summary><i>Previously, schandrika wrote…</i></summary><blockquote>

I will move it to common location
</blockquote></details>

Done.

---

*[volttrontesting/services/tagging/test_tagging.py, line 86 at r1](https://reviewable.io:443/reviews/volttron/volttron/1438#-Kr4tzXEbKQDh1o0Ghxw-r1-86:-KskPdmVA2vUtzqT-ksB:b-896fix) ([raw file](https://github.com/volttron/volttron/blob/a3a6903e590807b0d45649333cd43bef5b26f7ce/volttrontesting/services/tagging/test_tagging.py#L86)):*
<details><summary><i>Previously, schandrika wrote…</i></summary><blockquote>

It is getting set dynamically to volttron instance home in tagging_service fixture. Will remove the hardcoded path (that is not used ) from the declaration. 
</blockquote></details>

Done.

---

*[services/core/SQLiteTaggingService/sqlite/resources/categories.csv, line 1 at r1](https://reviewable.io:443/reviews/volttron/volttron/1438#-Kr4tzXEbKQDh1o0Ghxp-r1-1:-KskPgRIFaXBDBIN0UxJ:b-pww1wt) ([raw file](https://github.com/volttron/volttron/blob/a3a6903e590807b0d45649333cd43bef5b26f7ce/services/core/SQLiteTaggingService/sqlite/resources/categories.csv#L1)):*
<details><summary><i>Previously, schandrika wrote…</i></summary><blockquote>

I put it in two location since in theory they could be customized - adding new tags, categories etc. Also so that I could use resource_string method from package_resources
</blockquote></details>

Moved it to common location using __file__ to determine relative path. Updated and pushed changes

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/volttron/volttron/1438)*
<!-- Sent from Reviewable.io -->

schandrika commented:
Review status: 0 of 31 files reviewed at latest revision, 16 unresolved discussions.

---

*[services/core/SQLiteTaggingService/sqlite/resources/categories.csv, line 1 at r1](https://reviewable.io:443/reviews/volttron/volttron/1438#-Kr4tzXEbKQDh1o0Ghxp-r1-1:-KskPo2Sc6LlUqHHM0gi:b-896fix) ([raw file](https://github.com/volttron/volttron/blob/a3a6903e590807b0d45649333cd43bef5b26f7ce/services/core/SQLiteTaggingService/sqlite/resources/categories.csv#L1)):*
<details><summary><i>Previously, schandrika wrote…</i></summary><blockquote>

Moved it to common location using __file__ to determine relative path. Updated and pushed changes
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/volttron/volttron/1438)*
<!-- Sent from Reviewable.io -->

schandrika commented:
Review status: 0 of 31 files reviewed at latest revision, 16 unresolved discussions.

---

*[services/core/MongodbTaggingService/mongodb/tagging.py, line 102 at r1](https://reviewable.io:443/reviews/volttron/volttron/1438#-Kr4tzXEbKQDh1o0Ghxi-r1-102:-KskPv1umJFXdoKOA4d_:b-896fix) ([raw file](https://github.com/volttron/volttron/blob/a3a6903e590807b0d45649333cd43bef5b26f7ce/services/core/MongodbTaggingService/mongodb/tagging.py#L102)):*
<details><summary><i>Previously, schandrika wrote…</i></summary><blockquote>

removed
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/volttron/volttron/1438)*
<!-- Sent from Reviewable.io -->

craig8 commented:
Review status: 0 of 31 files reviewed at latest revision, 20 unresolved discussions.

---

*[services/core/MongodbTaggingService/mongodb/tagging.py, line 225 at r2](https://reviewable.io:443/reviews/volttron/volttron/1438#-KskV9ND8vL5ijzkkU9J:-KskV9NE4PmUjICfOd52:b-r7jbmj) ([raw file](https://github.com/volttron/volttron/blob/f9b71cb98c5917525d3e4dd990f57b1104635568/services/core/MongodbTaggingService/mongodb/tagging.py#L225)):*
> ```Python
>         file_path = self.resource_sub_dir+'/tags.csv'
>         _log.debug("Loading file :" + file_path)
>         with open(file_path, 'r') as content_file:
> ```

Probably shouldn't assume the user passes a valid path here.

---

*[services/core/MongodbTaggingService/mongodb/tagging.py, line 247 at r2](https://reviewable.io:443/reviews/volttron/volttron/1438#-KskVLSLFpwgiDIif9ix:-KskVLSLFpwgiDIif9iy:b-r7jbmj) ([raw file](https://github.com/volttron/volttron/blob/f9b71cb98c5917525d3e4dd990f57b1104635568/services/core/MongodbTaggingService/mongodb/tagging.py#L247)):*
> ```Python
>         with open(file_path, 'r') as content_file:
>             csv_str = content_file.read()
> ```

Probably shouldn't assume the user passes a valid path here.

---

*[services/core/MongodbTaggingService/mongodb/tagging.py, line 269 at r2](https://reviewable.io:443/reviews/volttron/volttron/1438#-KskVNczS7wTOz-Hhw0c:-KskVNczS7wTOz-Hhw0d:b-r7jbmj) ([raw file](https://github.com/volttron/volttron/blob/f9b71cb98c5917525d3e4dd990f57b1104635568/services/core/MongodbTaggingService/mongodb/tagging.py#L269)):*
> ```Python
>         with open(file_path, 'r') as content_file:
>             csv_str = content_file.read()
>         if csv_str:
> ```

Probably shouldn't assume the user passes a valid path here.

---

*[services/core/MongodbTaggingService/mongodb/tagging.py, line 284 at r2](https://reviewable.io:443/reviews/volttron/volttron/1438#-KskVPZJ1YSUyUC8_0gk:-KskVPZJ1YSUyUC8_0gl:b-r7jbmj) ([raw file](https://github.com/volttron/volttron/blob/f9b71cb98c5917525d3e4dd990f57b1104635568/services/core/MongodbTaggingService/mongodb/tagging.py#L284)):*
> ```Python
>         _log.debug("Loading file :" + file_path)
>         with open(file_path, 'r') as content_file:
>             txt_str = content_file.read()
> ```

Probably shouldn't assume the user passes a valid path here.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/volttron/volttron/1438)*
<!-- Sent from Reviewable.io -->

schandrika commented:
The path is not passed by user. Base_tagging agent sets the resource_sub_dir path to volttron_data/tagging. That is the common location from which both tagging agents are reading resource files now

From: Craig [mailto:notifications@github.com]
Sent: Tuesday, August 29, 2017 5:20 PM
To: VOLTTRON/volttron
Cc: Sivaramakrishnan, Chandrika; Mention
Subject: Re: [VOLTTRON/volttron] First version of Tagging Service Implementation (#1438)


Review status: 0 of 31 files reviewed at latest revision, 20 unresolved discussions.

________________________________

services/core/MongodbTaggingService/mongodb/tagging.py, line 225 at r2<https://reviewable.io:443/reviews/volttron/volttron/1438#-KskV9ND8vL5ijzkkU9J:-KskV9NE4PmUjICfOd52:b-r7jbmj> (raw file<https://github.com/volttron/volttron/blob/f9b71cb98c5917525d3e4dd990f57b1104635568/services/core/MongodbTaggingService/mongodb/tagging.py#L225>):

        file_path = self.resource_sub_dir+'/tags.csv'

        _log.debug("Loading file :" + file_path)

        with open(file_path, 'r') as content_file:

Probably shouldn't assume the user passes a valid path here.

________________________________

services/core/MongodbTaggingService/mongodb/tagging.py, line 247 at r2<https://reviewable.io:443/reviews/volttron/volttron/1438#-KskVLSLFpwgiDIif9ix:-KskVLSLFpwgiDIif9iy:b-r7jbmj> (raw file<https://github.com/volttron/volttron/blob/f9b71cb98c5917525d3e4dd990f57b1104635568/services/core/MongodbTaggingService/mongodb/tagging.py#L247>):

        with open(file_path, 'r') as content_file:

            csv_str = content_file.read()

Probably shouldn't assume the user passes a valid path here.

________________________________

services/core/MongodbTaggingService/mongodb/tagging.py, line 269 at r2<https://reviewable.io:443/reviews/volttron/volttron/1438#-KskVNczS7wTOz-Hhw0c:-KskVNczS7wTOz-Hhw0d:b-r7jbmj> (raw file<https://github.com/volttron/volttron/blob/f9b71cb98c5917525d3e4dd990f57b1104635568/services/core/MongodbTaggingService/mongodb/tagging.py#L269>):

        with open(file_path, 'r') as content_file:

            csv_str = content_file.read()

        if csv_str:

Probably shouldn't assume the user passes a valid path here.

________________________________

services/core/MongodbTaggingService/mongodb/tagging.py, line 284 at r2<https://reviewable.io:443/reviews/volttron/volttron/1438#-KskVPZJ1YSUyUC8_0gk:-KskVPZJ1YSUyUC8_0gl:b-r7jbmj> (raw file<https://github.com/volttron/volttron/blob/f9b71cb98c5917525d3e4dd990f57b1104635568/services/core/MongodbTaggingService/mongodb/tagging.py#L284>):

        _log.debug("Loading file :" + file_path)

        with open(file_path, 'r') as content_file:

            txt_str = content_file.read()

Probably shouldn't assume the user passes a valid path here.

________________________________

Comments from Reviewable<https://reviewable.io:443/reviews/volttron/volttron/1438>

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://github.com/VOLTTRON/volttron/pull/1438#issuecomment-325841066>, or mute the thread<https://github.com/notifications/unsubscribe-auth/APYL-Hzc_g6QufDtcYdgqlUmhFlYRALRks5sdKqvgaJpZM4Oxswn>.

craig8 commented:
Reviewed 2 of 41 files at r1, 11 of 21 files at r2.
Review status: 8 of 31 files reviewed at latest revision, 20 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/volttron/volttron/1438)*
<!-- Sent from Reviewable.io -->

craig8 commented:
Reviewed 1 of 41 files at r1.
Review status: 9 of 31 files reviewed at latest revision, 20 unresolved discussions, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/volttron/volttron/1438)*
<!-- Sent from Reviewable.io -->

craig8 commented:
Reviewed 3 of 41 files at r1, 1 of 21 files at r2.
Review status: 13 of 31 files reviewed at latest revision, 20 unresolved discussions, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/volttron/volttron/1438)*
<!-- Sent from Reviewable.io -->

schandrika commented:
renamed private methods. Fixed specification rst. 

---

Review status: 13 of 31 files reviewed at latest revision, 18 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/volttron/volttron/1438#-:-KskWOV3TWxYnhqnSBwC:blkvues)*
<!-- Sent from Reviewable.io -->
